econml/dml.py

```diff
@@ -335,28 +390,87 @@ class DMLCateEstimator(_DMLCateEstimatorBase):
+            else:
+                ols = OLS(y, X)
+            inference = self._inference  # Note: needs to be set on this instance from outside this class
+            if isinstance(inference, StatsModelsOptions):
+                self.results = ols.fit(cov_type=inference.cov_type, cov_kwds=inference.cov_kwds)
+            else:
+                self.results = ols.fit()
+            return self
+
+        def predict(self, X):
+            return self.results.predict(X)
+
+        def predict_interval(self, X, alpha):
+            # NOTE: we use `obs = False` to get a confidence, rather than prediction, interval
+            preds = self.results.get_prediction(X).conf_int(alpha=alpha, obs=False)
+            # statsmodels uses the last dimension instead of the first to store the confidence intervals,
+            # so we need to transpose the result
+            return transpose(preds)
+
     def __init__(self,
-                 model_y, model_t, model_final=LinearRegression(fit_intercept=False),
+                 model_y=LassoCV(), model_t=LassoCV(),
                  featurizer=PolynomialFeatures(degree=1, include_bias=True),
                  discrete_treatment=False,
                  n_splits=2,
                  random_state=None,
                  inference=None):
         super().__init__(model_y=model_y,
                          model_t=model_t,
-                         model_final=model_final,
+                         model_final=self.StatsModelsWrapper(),
                          featurizer=featurizer,
-                         sparseLinear=False,
+                         sparseLinear=True,
                          discrete_treatment=discrete_treatment,
                          n_splits=n_splits,
                          random_state=random_state,
                          inference=inference)
 
+    def _fit_impl(self, *args, **kwargs):
+        # TODO: is there a cleaner way to achieve this?
+        # Note that this assignment needs to be done after the super call, so that
+        # self._inference is correctly set
+        # Also, the instantiated StatsModelWrapper is cloned (so we can't set the property on the original model)
+        # and for bootstrap, the instance will be cloned by the base class (so we can't set the property at the end
+        # of the initializer, either, or it will only apply to the non-cloned instances)
+        self._model_final._model._inference = self._inference
+        return super()._fit_impl(*args, **kwargs)
+
+    def _effect_interval_impl(self, X, *, T0=0, T1=1, alpha=0.1):
+        if isinstance(self._inference, StatsModelsOptions):
+            dT = T1 - T0
+            preds = self._model_final._model.predict_interval(self._model_final._combine(X, dT), alpha=alpha)
+            return preds
+        else:
+            return super()._effect_interval_impl(X, T0=T0, T1=T1, alpha=alpha)
+
+    def const_marginal_effect_interval(X, *, alpha=0.1):
```

Comment:
It's a bit confusing that we sometime implement the original method of the parent class, and other times we implement the _impl method. Also this is not inline with what we are doing in all our other estimators outside of dml. 

Also is there a way to not have to do sth different for each inference method? I was hoping that all the logic of the inference method would be carried out inside the inference class and nothing should change in the estimator class. What if I want to add 10 more inference methods for the dml class. Would I need to write 10 if then else statements inside of _effect_interval_impl?

Reply:
Fixed.